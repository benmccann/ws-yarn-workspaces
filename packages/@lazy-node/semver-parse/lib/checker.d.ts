import { IHasOperator, ISimpleSemVerObjectBase } from './types';
export declare function isSimpleSemVerOperatorLike<T extends ISimpleSemVerObjectBase>(obj: T): obj is IHasOperator<T> & {
    major?: never;
};
export declare function assertSimpleSemVerOperatorLike<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is IHasOperator<T> & {
    major?: never;
};
export declare function isSimpleSemVerObjectLike<T extends ISimpleSemVerObjectBase>(obj: T): obj is T & {
    major: string;
};
export declare function assertSimpleSemVerObjectLike<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is T & {
    major: string;
};
export declare function hasOperator<T extends ISimpleSemVerObjectBase>(obj: T): obj is IHasOperator<T>;
export declare function assertHasOperator<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is IHasOperator<T>;
