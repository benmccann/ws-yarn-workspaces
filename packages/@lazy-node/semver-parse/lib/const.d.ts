export declare const reSemver: RegExp;
export declare const reSemverWithRange: RegExp;
/**
 * , reSemverRange = /\s*((\|\||\-)|(([<>~]?=?)\s*(v)?([0-9]+)(\.(x|[0-9]+))?(\.(x|[0-9]+))?(([\-+])([a-zA-Z0-9\.]+))?))\s* /g
 */
export declare const reSemverRange: RegExp;
export declare const simpleSemVerKeys: readonly ["semver", "operator", "version", "major", "minor", "patch", "release", "build"];
export declare const enum EnumVersionExtra {
    build = "+",
    release = "-"
}
