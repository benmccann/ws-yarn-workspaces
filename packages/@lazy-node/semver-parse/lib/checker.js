"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.assertHasOperator = exports.hasOperator = exports.assertSimpleSemVerObjectLike = exports.isSimpleSemVerObjectLike = exports.assertSimpleSemVerOperatorLike = exports.isSimpleSemVerOperatorLike = void 0;
function isSimpleSemVerOperatorLike(obj) {
    var _a, _b;
    return (!((_a = obj.major) === null || _a === void 0 ? void 0 : _a.length) && ((_b = obj.operator) === null || _b === void 0 ? void 0 : _b.length) > 0);
}
exports.isSimpleSemVerOperatorLike = isSimpleSemVerOperatorLike;
function assertSimpleSemVerOperatorLike(obj, notThrow) {
    if (notThrow !== true && !isSimpleSemVerOperatorLike(obj)) {
        throw new TypeError(`obj not a SimpleSemVerOperator`);
    }
}
exports.assertSimpleSemVerOperatorLike = assertSimpleSemVerOperatorLike;
function isSimpleSemVerObjectLike(obj) {
    var _a;
    return ((_a = obj.major) === null || _a === void 0 ? void 0 : _a.length) > 0;
}
exports.isSimpleSemVerObjectLike = isSimpleSemVerObjectLike;
function assertSimpleSemVerObjectLike(obj, notThrow) {
    if (notThrow !== true && !isSimpleSemVerObjectLike(obj)) {
        throw new TypeError(`obj not a SimpleSemVerObject`);
    }
}
exports.assertSimpleSemVerObjectLike = assertSimpleSemVerObjectLike;
function hasOperator(obj) {
    var _a;
    return ((_a = obj.operator) === null || _a === void 0 ? void 0 : _a.length) > 0;
}
exports.hasOperator = hasOperator;
function assertHasOperator(obj, notThrow) {
    if (notThrow !== true && !hasOperator(obj)) {
        throw new TypeError(`obj not has operator`);
    }
}
exports.assertHasOperator = assertHasOperator;
//# sourceMappingURL=checker.js.map