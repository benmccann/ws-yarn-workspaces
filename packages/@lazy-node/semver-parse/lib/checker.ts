import {
	ISimpleSemVer,
	ISimpleSemVerOperator,
	ISimpleSemVerObject,
	IHasOperator,
	ISimpleSemVerObjectBase,
} from './types';

export function isSimpleSemVerOperatorLike<T extends ISimpleSemVerObjectBase>(obj: T): obj is IHasOperator<T> & {
	major?: never,
}
{
	return (!obj.major?.length && obj.operator?.length > 0)
}

export function assertSimpleSemVerOperatorLike<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is IHasOperator<T> & {
	major?: never,
}
{
	if (notThrow !== true && !isSimpleSemVerOperatorLike(obj))
	{
		throw new TypeError(`obj not a SimpleSemVerOperator`)
	}
}

export function isSimpleSemVerObjectLike<T extends ISimpleSemVerObjectBase>(obj: T): obj is T & {
	major: string
}
{
	return obj.major?.length > 0
}

export function assertSimpleSemVerObjectLike<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is T & {
	major: string
}
{
	if (notThrow !== true && !isSimpleSemVerObjectLike(obj))
	{
		throw new TypeError(`obj not a SimpleSemVerObject`)
	}
}

export function hasOperator<T extends ISimpleSemVerObjectBase>(obj: T): obj is IHasOperator<T>
{
	return obj.operator?.length > 0
}

export function assertHasOperator<T extends ISimpleSemVerObjectBase>(obj: T, notThrow?: boolean): asserts obj is IHasOperator<T>
{
	if (notThrow !== true && !hasOperator(obj))
	{
		throw new TypeError(`obj not has operator`)
	}
}
