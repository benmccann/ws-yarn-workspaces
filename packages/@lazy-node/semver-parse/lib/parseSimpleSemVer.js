"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.parseSimpleSemVer = void 0;
const const_1 = require("./const");
const SimpleSemVer_1 = require("./SimpleSemVer");
const checker_1 = require("./checker");
function parseSimpleSemVer(version) {
    // semver, major, minor, patch
    // https://github.com/mojombo/semver/issues/32
    // https://github.com/isaacs/node-semver/issues/10
    // optional v
    const m = const_1.reSemverWithRange.exec(version);
    let ver;
    if ((m === null || m === void 0 ? void 0 : m.length) > 0) {
        let [semver, operator, version, major, minor, patch, release, build] = m;
        ver = new SimpleSemVer_1.SimpleSemVer({
            operator,
            semver,
            version,
            major,
            minor,
            patch,
            release,
            build,
        });
        checker_1.assertSimpleSemVerObjectLike(ver);
    }
    return ver;
}
exports.parseSimpleSemVer = parseSimpleSemVer;
exports.default = parseSimpleSemVer;
//# sourceMappingURL=parseSimpleSemVer.js.map