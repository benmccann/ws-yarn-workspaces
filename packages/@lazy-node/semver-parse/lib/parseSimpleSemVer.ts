import { reSemver, reSemverWithRange } from './const';
import { SimpleSemVer } from './SimpleSemVer';
import { pruned } from './util/pruned';
import { ISimpleSemVer } from './types';

export function parseSimpleSemVer<T extends ISimpleSemVer = ISimpleSemVer>(version: string)
{
	// semver, major, minor, patch
	// https://github.com/mojombo/semver/issues/32
	// https://github.com/isaacs/node-semver/issues/10
	// optional v
	const m = reSemverWithRange.exec(version);
	let ver: SimpleSemVer;

	if (m?.length > 0)
	{
		let [semver, operator, version, major, minor, patch, release, build] = m;

		ver = new SimpleSemVer({
			operator,
			semver,
			version,
			major,
			minor,
			patch,
			release,
			build,
		})
	}

	return ver;
}

export default parseSimpleSemVer
