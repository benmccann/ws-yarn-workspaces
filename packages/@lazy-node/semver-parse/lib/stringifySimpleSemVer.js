"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.stringifySemverFull = exports.stringifySimpleSemVer = void 0;
const checker_1 = require("./checker");
function stringifySimpleSemVer(obj) {
    var _a, _b, _c, _d, _e;
    (0, checker_1.assertSimpleSemVerObjectLike)(obj);
    let str = '';
    str += (_a = obj.major) !== null && _a !== void 0 ? _a : '0';
    str += '.';
    str += (_b = obj.minor) !== null && _b !== void 0 ? _b : '0';
    str += '.';
    str += (_c = obj.patch) !== null && _c !== void 0 ? _c : '0';
    if (((_d = obj.release) === null || _d === void 0 ? void 0 : _d.length) > 0) {
        str += "-" /* EnumVersionExtra.release */ + obj.release;
    }
    if (((_e = obj.build) === null || _e === void 0 ? void 0 : _e.length) > 0) {
        str += "+" /* EnumVersionExtra.build */ + obj.build;
    }
    return str;
}
exports.stringifySimpleSemVer = stringifySimpleSemVer;
function stringifySemverFull(obj) {
    var _a;
    return ((_a = obj.operator) !== null && _a !== void 0 ? _a : '') + stringifySimpleSemVer(obj);
}
exports.stringifySemverFull = stringifySemverFull;
exports.default = stringifySimpleSemVer;
//# sourceMappingURL=stringifySimpleSemVer.js.map