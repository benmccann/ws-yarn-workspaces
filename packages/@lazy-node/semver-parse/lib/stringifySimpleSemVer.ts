import { ISimpleSemVerObject } from './types';
import { EnumVersionExtra } from './const';
import SimpleSemVer from './SimpleSemVer';
import { assertSimpleSemVerOperatorLike, assertSimpleSemVerObjectLike } from './checker';

export function stringifySimpleSemVer(obj: ISimpleSemVerObject | SimpleSemVer)
{
	assertSimpleSemVerObjectLike(obj);

	let str = '';

	str += obj.major ?? '0';
	str += '.';
	str += obj.minor ?? '0';
	str += '.';
	str += obj.patch ?? '0';

	if (obj.release?.length > 0)
	{
		str += EnumVersionExtra.release + obj.release;
	}

	if (obj.build?.length > 0)
	{
		str += EnumVersionExtra.build + obj.build;
	}

	return str;
}

export function stringifySemverFull(obj: ISimpleSemVerObject | SimpleSemVer)
{
	return (obj.operator ?? '') + stringifySimpleSemVer(obj)
}

export default stringifySimpleSemVer

