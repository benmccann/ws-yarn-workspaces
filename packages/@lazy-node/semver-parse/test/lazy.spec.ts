import parseSimpleSemVer from '../lib/parseSimpleSemVer';
import { isSimpleSemVerObjectLike, hasOperator, assertSimpleSemVerObjectLike } from '../lib/checker';
import { stringifySemverFull } from '../lib/stringifySimpleSemVer';

describe(`parseSimpleSemVer`, () =>
{

	test(`^4.0.0-dev.20200615`, () =>
	{
		let version = `^4.0.0-dev.20200615`;

		let actual = parseSimpleSemVer(version);

		expect(isSimpleSemVerObjectLike(actual)).toBeTruthy();
		expect(hasOperator(actual)).toBeTruthy();

		expect(stringifySemverFull(actual)).toStrictEqual(version);

		expect(actual).toMatchSnapshot();

	});

})
