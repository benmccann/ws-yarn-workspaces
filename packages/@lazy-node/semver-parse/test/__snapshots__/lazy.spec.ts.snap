// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`parseSimpleSemVer ^4.0.0-dev.20200615 1`] = `
Object {
  "major": "4",
  "minor": "0",
  "operator": "^",
  "patch": "0",
  "release": "dev.20200615",
  "semver": "^4.0.0-dev.20200615",
  "version": "4.0.0",
}
`;
